help_to_heat/utils.py

```diff
@@ -249,3 +250,17 @@ def get_current_and_next_month_names(month_names):
 # this is to tell the user the first month we have no EPCs for, and the next month we expect to perform a dump
 def get_current_scottish_epc_cutoff_and_next_dump_month_names(month_names):
     return month_names[9], month_names[1]
+
+
+def get_most_recent_epc_per_uprn(address_and_lmk_details):
+    most_recent_address_and_lmk_details = []
+    sorted_address_and_lmk_details = sorted(address_and_lmk_details, key=lambda x: x["uprn"], reverse=True)
+
+    for uprn, group in itertools.groupby(sorted_address_and_lmk_details, lambda x: x["uprn"]):
```

Comment:
I suppose it's not clear at this time whether a missing UPRN will come through as an empty string or be vacant entirely; probably safer to use the `.get()` method rather than indexing to provide a sensible default & avoid an error being thrown (probably `""` to align with the below line)

probably safer to use `.get()` in all cases here